src/com/rcv/Main.java

```diff
@@ -45,53 +45,63 @@ public static void main(String[] args) {
       String configPath = args[0];
       // config file for running the tabulator
       ElectionConfig config = loadElectionConfig(configPath);
-      Logger.info("Tabulator is being used via the CLI.");
-      executeTabulation(config);
+      if (config != null) {
+        Logger.info("Tabulator is being used via the CLI.");
+        executeTabulation(config);
+      } else {
+        Logger.severe("Aborting because config is invalid.");
+      }
     }
   }
 
   // function: loadElectionConfig
   // purpose: create config object
   // param: path to config file
-  // returns: the new ElectionConfig object or null if there was a problem
+  // returns: the new ElectionConfig object, or null if there was a problem
   static ElectionConfig loadElectionConfig(String configPath) {
     // config: the new object
     ElectionConfig config = null;
 
-    // encounteredError tracks whether anything went wrong during loading
-    boolean encounteredError = false;
+    // tracks whether we encountered any critical file system errors
+    boolean encounteredFileError = false;
 
     // rawConfig holds the basic election config data parsed from json
     RawElectionConfig rawConfig =
         JsonParser.parseObjectFromFile(configPath, RawElectionConfig.class);
 
     if (rawConfig == null) {
-      encounteredError = true;
+      System.err.println(String.format("Failed to load config file: %s", configPath));
     } else {
       config = new ElectionConfig(rawConfig);
 
       try {
         FileUtils.createOutputDirectory(config.getOutputDirectory());
       } catch (UnableToCreateDirectoryException exception) {
-        encounteredError = true;
         System.err.println(
             String.format(
                 "Failed to create output directory: %s\n%s",
                 config.getOutputDirectory(),
                 exception.toString()
             )
         );
+        encounteredFileError = true;
       }
     }
 
-    // TODO: confiog.validate() should log specific errors
-    if (encounteredError || !config.validate()) {
-      Logger.severe("There was a problem loading or validating the election configuration.");
-      Logger.severe("Please see the README.txt for details.");
+    if (config != null) {
```

Comment:
same thing here

Reply:
The way this method is written, we do need this check... but I'll restructure the logic in the method slightly so it's not necessary.